### ISSUE TITLE
`module_name_repetitions` doesn't warn if the prefix/suffix is the same as the identifier.

### ISSUE THREAD

**Issue by SLUCHABLUB** [rust-lang/rust-clippy#14223]
### Summary

If you define an item with the same name as its parent module, the `module_name_repetitions` lint doesn't warn. The documentation states: "Detects public item names that are prefixed or suffixed by the containing public module’s name." and according to rust, a string is prefixed/suffixed by itself.

If this is the intended behavior, I believe that the documentation should be adjusted and perhaps a new lint added.

### Lint Name

module_name_repetitions

### Reproducer

I tried this code:

```rust
#![warn(clippy::module_name_repetitions)]

pub mod foo {
    // Warns
    pub struct BarFoo;
    // Warns
    pub struct FooBar;
    // Doesn't warn
    pub struct Foo;
}
```

I expected to see this happen:

Getting a warning for the item `foo::Foo`.

Instead, this happened:

I got no warning.

### Version

```text
rustc 1.84.1 (e71f9a9a9 2025-01-27)
binary: rustc
commit-hash: e71f9a9a98b0faf423844bf0ba7438f29dc27d58
commit-date: 2025-01-27
host: aarch64-apple-darwin
release: 1.84.1
LLVM version: 19.1.5
```

**Comment by samueltardieu** [comment]
It is quite common to define a name `Foo` in a module `foo`. In plain English, as explained, a prefix or suffix is added to the beginning or the end of a name, but cannot constitute the entire name. I am not sure if there is a need to fix the documentation in that regard.

**Comment by SLUCHABLUB** [comment]
Understandable, personally I find `foo::Foo` equally unaesthetic as it in the same manner requires you to repeat the module name. Since it doesn't seem to be the scope of the lint however, I'll make a feature request instead.
